cistem/protocols/protocol_tomo_resample.py

```diff
@@ -0,0 +1,184 @@
+        # New Z-Size [232]                                   : 232
+
+
+        # You need a params to belong to a section:
+        form.addSection(label=Message.LABEL_INPUT)
+        form.addParam('inTomograms', PointerParam,
+                        pointerClass='SetOfTomograms',
+                        allowsNull=False,
+                        label='Input tomograms')
+
+        form.addParam('newXsize', IntParam,
+                        default=512,
+                        label='New X-Size',
+                        help='Volume will be rescaled to this size in X dimension (voxels)')
+        form.addParam('newYsize', IntParam,
+                        default=512,
+                        label='New Y-Size',
+                        help='Volume will be rescaled to this size in Y dimension (voxels)')
+        form.addParam('newZsize', IntParam,
+                        default=256,
+                        label='New Z-Size',
+                        help='Volume will be rescaled to this size in Z dimension (voxels)')
+
+    # -------------------------- INSERT steps functions -----------------------
+    def _insertAllSteps(self):
+
+        for tomo in self.inTomograms.get():
+
+            self._insertFunctionStep(self.runTomoResample,
+                        tomo.getFileName())
+
+        self._insertFunctionStep(self.createOutputStep)
+
+    def runTomoResample(self, tomoFile: str):
+
+        prog = Plugin.getProgram('resample')
+
+
+        tomoBaseName = removeBaseExt(tomoFile)
+        tomoExt = getExt(tomoFile)
+        paramDict = {}
+        paramDict['tomoFile'] = tomoFile
+        paramDict['tomoOutName'] = self.getWorkingDir() + '/' + OUTPUT_DIR + '/' + \
+            tomoBaseName + '_resampled' + tomoExt
+        
+        paramDict['newXsize'] = self.newXsize
+        paramDict['newYsize'] = self.newYsize
+        paramDict['newZsize'] = self.newZsize
+
+        # Arguments to the resample command defined in the plugin initialization:
+        args = """   << eof
+%(tomoFile)s
+%(tomoOutName)s
+YES
+%(newXsize)d
+%(newYsize)d
+%(newZsize)d
+eof\n
+"""
+        self.runJob(prog, args % paramDict)
+
+        self.tomoList.append(paramDict['tomoOutName'])
+
+    def createOutputStep(self):
+        labelledSet = self._genOutputSetOfTomograms(
+            self.tomoList, 'resampled')
+        self._defineOutputs(**{OUTPUT_TOMO_NAME: labelledSet})
+        self._defineSourceRelation(self.inTomograms.get(), labelledSet)
+
+    def _genOutputSetOfTomograms(self, tomoList, suffix):
+        tomoSet = SetOfTomograms.create(
+            self._getPath(), template='tomograms%s.sqlite', suffix=suffix)
+        inTomoSet = self.inTomograms.get()
+        tomoSet.copyInfo(inTomoSet)
+
+        outSamplingRate = Ccp4Header(tomoList[0], readHeader=True).getSampling()
+        tomoSet.setSamplingRate(outSamplingRate[0])
+
+        counter = 1
+        for file, inTomo in zip(tomoList, inTomoSet):
```

Comment:
while this will work, it is better to use copyItems method from EMSet. You can find an example in deconv tomo protocol in the warp plugin, e.g. out_tomos.copyItems(in_tomos, updateItemCallback=self._updateItem). I can help with this